src/vs/workbench/contrib/chat/browser/languageModelToolsService.ts

```diff
@@ -91,6 +97,17 @@ export class LanguageModelToolsService extends Disposable implements ILanguageMo
 		this._ctxToolsCount = ChatContextKeys.Tools.toolsCount.bindTo(_contextKeyService);
 	}
 
+	acceptProgress(sessionId: string | undefined, callId: string, progress: IProgressStep): void {
```

Comment:
This is not nice and looks like "IPC leaks in to service interface". I would expect that `invokeTool` must now be called with progress and that the "accept-progress-bits-for-call"-logic is something that's an implementation detail of `mainThreadLanguageModelTools`